lib/topics.js

```diff
@@ -8,18 +8,37 @@ const SNS    = require('./sns');
 class Topic {
 
   constructor(server, topicName) {
-    this._server = server;
-    this.name    = topicName;
+    this._server     = server;
+    this.name        = topicName;
+    this.subscribers = [];
   }
 
 
   // Publish to a topic.
   publish(message) {
-    return this.clientPromise
-      .then(client => {
-        const stringMessage = JSON.stringify(message);
-        return client.publish(this.name, stringMessage);
+    const stringMessage = JSON.stringify(message);
+
+    if (process.env.NODE_ENV === 'test') {
+      // JSON encode/decode the message before passing to
+      // subscriber, same transformation as when using SNS.
+      const parsedMessage = JSON.parse(stringMessage);
+      setImmediate(() => {
+        this.subscribers.forEach(handler => handler(parsedMessage));
       });
+      return Promise.resolve();
+    } else {
+      return this.clientPromise
+        .then(client => client.publish(this.name, stringMessage));
+    }
```

Comment:
Say you're showing a real time activity stream in the browser, new activities get added via notification. Each user has their own timeline, but only active users care to receive notifications.

(* BTW not a good use case for SNS at the moment, the protocols they use are not optimized for that)